tests/devices/unit_tests/test_xspress3.py

```diff
@@ -0,0 +1,78 @@
+import asyncio
+from unittest.mock import AsyncMock
+
+import bluesky.plan_stubs as bps
+import pytest
+from bluesky.run_engine import RunEngine
+from ophyd_async.core import DeviceCollector, callback_on_mock_put, set_mock_value
+
+from dodal.devices.xspress3.xspress3 import (
+    AcquireRBVState,
+    AcquireState,
+    AsyncStatus,
+    DetectorState,
+    TriggerMode,
+    Xspress3,
+)
+
+
+@pytest.fixture
+async def mock_xspress3mini(prefix: str = "BLXX-EA-DET-007:") -> Xspress3:
+    async with DeviceCollector(mock=True):
+        mock_xspress3mini = Xspress3(prefix, "Xspress3Mini", 2)
+    assert mock_xspress3mini.channels[1].name == "Xspress3Mini-channels-1"
+    assert mock_xspress3mini.channels[2].name == "Xspress3Mini-channels-2"
+    assert (
+        mock_xspress3mini.get_roi_calc_status[1].name
+        == "Xspress3Mini-get_roi_calc_status-1"
+    )
+    assert (
+        mock_xspress3mini.get_roi_calc_status[2].name
+        == "Xspress3Mini-get_roi_calc_status-2"
+    )
+    assert mock_xspress3mini.roi_mca[1].name == "Xspress3Mini-roi_mca-1"
+    assert mock_xspress3mini.roi_mca[2].name == "Xspress3Mini-roi_mca-2"
+    mock_xspress3mini.timeout = 1
+    return mock_xspress3mini
+
+
+async def test_stage_success_on_busy_state(mock_xspress3mini: Xspress3):
+    set_mock_value(mock_xspress3mini.detector_state, DetectorState.ACQUIRE)
+    callback_on_mock_put(
+        mock_xspress3mini.acquire,
+        lambda _: set_mock_value(
+            mock_xspress3mini.acquire_rbv, AcquireRBVState.ACQUIRE
+        ),
+    )
+
+    mock_xspress3mini.trigger_mode.set = AsyncMock()
+    mock_xspress3mini.acquire.set = AsyncMock()
+    status: AsyncStatus = mock_xspress3mini.stage()
+
+    assert status.done is False
+    while not status.done:
+        await asyncio.sleep(0.1)
+    assert status.done is True
+
+    mock_xspress3mini.trigger_mode.set.assert_called_once_with(TriggerMode.BURST)
+    mock_xspress3mini.acquire.set.assert_called_once_with(AcquireState.ACQUIRE)
```

Comment:
Must: This test should actually be failing. Your `stage` waits for the RBV to go to `DONE` but you have no callback in this test that sets it to `DONE`. Additionally:
* You can use the in-built `get_mock_put` rather than setting your own mocks
* You can just `await ... .stage` to call it:
```suggestion
async def test_stage_success_on_busy_state(mock_xspress3mini: Xspress3):
    set_mock_value(mock_xspress3mini.detector_state, DetectorState.ACQUIRE)
    callback_on_mock_put(
        mock_xspress3mini.acquire,
        lambda *_, **__: set_mock_value(
            mock_xspress3mini.acquire_rbv, AcquireRBVState.ACQUIRE
        ),
    )
    await mock_xspress3mini.stage()

    get_mock_put(mock_xspress3mini.trigger_mode).assert_called_once_with(
        TriggerMode.BURST, wait=ANY, timeout=ANY
    )
    get_mock_put(mock_xspress3mini.acquire).assert_called_once_with(
        AcquireState.ACQUIRE, wait=ANY, timeout=ANY
    )
```